packages/pros/src/adi/analog.rs

```diff
@@ -0,0 +1,98 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use crate::error::bail_on;
+
+use pros_sys::PROS_ERR;
+
+pub struct AdiAnalogIn {
+    port: u8,
+}
+
+impl AdiAnalogIn {
+    /// Create an AdiAnalogIn, returning err `AdiError::InvalidPort` if the port is invalid.
+    pub fn new(slot: AdiSlot) -> Result<Self, AdiError> {
+        let port = slot as u8;
+        if port < 1 || port > {pros_sys::NUM_ADI_PORTS as u8} {
+            return Err(AdiError::InvalidPort);
```

Comment:
This check seems unreacahble, since AFAIK casting an `AdiSlot` to u8 will always be in this range.